### ISSUE TITLE
no-extra-parens ignores param and destructuring defaults

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#11893]
**Tell us about your environment**

* **ESLint Version:**  6.0.0 (same with 5.16.0)
* **Node Version:** 10.16.0
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parserOptions: {
    ecmaVersion: 6,
  },
  rules: {
    "no-extra-parens": "error",
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function foo(a = (b)) {}

const bar = (a = (b)) => a;

const [a = (b)] = [];

const {c = (d)} = {};
```

```bash
eslint index.js
```

**What did you expect to happen?**

4 errors.

**What actually happened? Please include the actual, raw output from ESLint.**

Nothing, no errors.

**Are you willing to submit a pull request to fix this bug?**

Yes, I'll do it.

Inner expressions are fine (e.g. `((b * c) + d))` would be reported), only the top level is missing.

Found this while working on [#11848](https://github.com/eslint/eslint/pull/11848), the `"AssignmentPattern"` handler is missing. It should check its `.right`.

Just need help with the analysis:

`comma` operator (`"SequenceExpression"`) inside should be checked for sure.

Is there any other case where these parens could be necessary?

**Comment by mysticatea** [comment]
Thank you for your report.

I confirmed it.
